### ISSUE TITLE
All Gmail accounts sign out in the browser run in Sandboxie after 1-2 days.

### ISSUE THREAD

**Issue by W1K23** [sandboxie-plus/Sandboxie#3686]
### Describe what you noticed and did

It started last summer. I thought it was a Gmail, browser, or email downloader addon issue, as it began signing out of all my accounts (including family members') from the browser approximately every 1-2 days. It became increasingly annoying, but I couldn't figure out what was causing the problem, as it had never signed out accounts with the same browser settings for many years before. Cookies were enabled, etc. Meanwhile, I switched browsers, and recently I discovered that it only signs out accounts when I run the browser in Sandboxie. I've been signed into the accounts with the other browser for about 2 months, but I haven't started that browser in Sandboxie since. I realized that Sandboxie is causing this annoying issue. What could be the problem? Which setting is the cause?

### Expected behavior

Keep the Gmail accounts logged in, even in the browser launched in Sandboxie.

### Affected program

Firefox, Vivaldi, Edge

### Download link

Not relevant

### Where is the program located?

The program is installed only outside the sandbox.

### Did the program or any related process close unexpectedly?

No, not at all.

### What version of Sandboxie are you running now?

Plus 1.12.9

### Is it a new installation of Sandboxie?

I just updated Sandboxie from a previous version (I remember which one it is).

### In which sandbox type you have this problem?

In a standard isolation sandbox (yellow sandbox icon).

### Can you reproduce this problem on a new empty sandbox?

I can confirm it also on a new empty sandbox.

### What is your Windows edition and version?

Win 10 Pro

### In which Windows account you have this problem?

Not relevant to my request.

### Please mention any installed security software

Not relevant

**Comment by e-t-l** [comment]
I can't replicate this. 

Any additional windows settings/programs that could possibly affect it, like a VPN? Can you replicate the behavior on a new device, in a different user profile, or in a virtual machine like VMware?

**Comment by W1K23** [comment]
> I can't replicate this.
> 
> Any additional windows settings/programs that could possibly affect it, like a VPN? Can you replicate the behavior on a new device, in a different user profile, or in a virtual machine like VMware?

Hi! The same issue is occurring on multiple computers, not only for me but also for family members. We are not using a VPN, VBM. I'm not aware of any program or setting that could be affecting this.

**Comment by W1K23** [comment]
> Could you please share your Sandboxie configuration?

Enabled=y
BlockNetworkFiles=y
RecoverFolder=%{374DE290-123F-4565-9164-39C4925E467B}%
RecoverFolder=%Personal%
RecoverFolder=%Desktop%
RecoverFolder=D:\
BorderColor=#00ffff,ttl,6
Template=OpenBluetooth
Template=SkipHook
Template=FileCopy
Template=qWave
Template=BlockPorts
Template=LingerPrograms
Template=AutoRecoverIgnore
Template=Firefox_Bookmarks_DirectAccess
Template=Vivaldi_Bookmarks_DirectAccess
Template=Chrome_Bookmarks_DirectAccess
Template=Edge_Bookmarks_DirectAccess
ConfigLevel=10
AutoDelete=y
AutoRecover=y

**Comment by offhub** [comment]
Automatic content deletion is enabled (`AutoDelete=y`) for the sandbox. If you only sign in to your Gmail account from the sandbox, your browser settings will also be deleted when the sandbox content is deleted.

**Comment by W1K23** [comment]
> Automatic content deletion is enabled (`AutoDelete=y`) for the sandbox. If you only sign in to your Gmail account from the sandbox, your browser settings will also be deleted when the sandbox content is deleted.

No. I sign into the Gmail accounts in the browser launched in the usual way. (I also perform updates and settings at that time.) Afterward, I start the browser in Sandboxie.

**Comment by W1K23** [comment]
For years I used Firefox with the X-notifier addon (on multiple computers) launched in Sandboxie. When it started signing out of Gmail accounts at first, I thought it was an issue with the addon. Therefore I replaced Xnotifier with other Gmail checking addons, but it still signed them out. That's when I suspected Firefox.
I switched to another browser, Vivaldi and use it in Sandboxie. After a few days something in it also signs out the Gmail accounts after I close the browser.
Currently I use 3-4 different browsers in normal mode and only one run in Sandboxie. For months none of the browsers signed out Gmail accounts except the one launched in Sandboxie.

**Comment by W1K23** [comment]
For years I used Firefox with the X-notifier addon (on multiple computers) launched in Sandboxie. When it started signing out of Gmail accounts at first, I thought it was an issue with the addon. I replaced Xnotifier with other Gmail checking addons, but it still signed Gmails out. Then I thought it might be a Firefox issue.
I switched to another browser, Vivaldi and use it in Sandboxie. After a few days something in it also signs out the Gmail accounts after I close the browser.
Currently I use 3-4 different browsers in normal mode and only one run in Sandboxie. For months none of the browsers have signed out Gmail accounts except the one that is running in Sandboxie.

**Comment by offhub** [comment]
Have you tried the opposite?
1. Login to Gmail with unsandboxed browser
2. Disable automatic deletion in the sandbox where you use the browser (`AutoDelete=n`)
3. Open the browser once in the sandbox, use Gmail, and close it
4. After using the browser outside the sandbox for a few days, open the browser in the sandbox again and open the Gmail page.
5. See if the account is logged out this way in the sandbox.

**Comment by W1K23** [comment]
Further developments. I launched the browser, logged into all Gmail accounts. I closed the browser and then launched it in Sandboxie. Gmail accounts were logged in. I closed the browser. I can start that browser anytime outside of Sboxie and the Gmail accounts have been logged in for over a week now.
I performed this operation with another browser as well, but I constantly run it in Sandboxie. If I close and reopen the browser which runs in Sboxie within 1 day the Gmail accounts remain logged in, but if I close it for more than 1-2 days, it logs out all of them. Meanwhile the other browser which is open outside of Sboxie, keeps the Gmail accounts logged in indefinitely. Something happens in Sboxie, not within 1 day, but rather after 2-3 days. It's not because of automatic content deletion because that happens after every closure and I can close it as many times as I want within 1 day (maybe 2) without logging out the accounts. Since last summer I haven't been able to figure out what could be causing this. Never happened before. I've been using Sandboxie and browsers with the same settings for years.

**Comment by offhub** [comment]
Before you run Firefox normally (after 1-2 days), check to see if the files in Firefox's profile folder have changed since it was last used. (except for the following files) If files other than those listed below are changed, a third-party cleaning utility may be causing this problem.

```
%APPDATA%\Mozilla\Firefox\favicons.sqlite
%APPDATA%\Mozilla\Firefox\favicons.sqlite-shm
%APPDATA%\Mozilla\Firefox\favicons.sqlite-wal
%APPDATA%\Mozilla\Firefox\places.sqlite
%APPDATA%\Mozilla\Firefox\places.sqlite-shm
%APPDATA%\Mozilla\Firefox\places.sqlite-wal

%APPDATA%\Mozilla\bookmarkbackups\ (all files under this folder)
```

**Comment by W1K23** [comment]
For years Firefox has been my main browser which I ran in Sandboxie with the same settings for years. I've never had any issues with Sandboxie. It never logged me out of logged-in accounts, forums, various sites, my Gmail accounts remained logged in continuously, whether I launched the browser in Sandboxie or outside of it. Last July it started increasingly logging out my Gmail accounts when running the browser in Sboxie. Other sites weren't affected, only Gmail accounts. The same thing happened on three different computers, all with the same settings. We don't use any cleaning programs or add-ons. Cookies are enabled. Last November I had had enough because Firefox running in Sandboxie disables HW acceleration, so the computer doesn't use the GPU and consumes much more CPU, causing the laptop to heat up significantly, while there's no issue with hardware acceleration. I asked the developer to do something about it because there hadn't been any problems before, but He didn't care. Interestingly Chromium browsers have HW acceleration even in Sandboxie. 
Then came the Gmail logout problem every 1-2 days. That's why Vivaldi became my main browser since November. Hardware acceleration works fine in it, but unfortunately the Gmail logout issue arose there too. Meanwhile I also used Firefox, but only running it outside of Sandboxie. Since then every Gmail acc in Firefox has remained logged in and it hasn't logged out any of them for months! That's when I realized Sandboxie was causing the logout. I don't know what Sandboxie and Gmail are related to. 
**In recent days I started using Firefox again running it in Sandboxie. I launch Vivaldi outside of Sandboxie. After 2 days I closed Firefox and then every Gmail account was logged out. In Vivaldi (outside of SB) though every Gmail acc remains logged in.** 
**It's clearly a Sandboxie problem.** I just can't figure out what it is. :(

**Comment by W1K23** [comment]
> As it has been 3 months since the last activity, we are automatically closing this issue in 14 days. If it is still present, please respond to help us investigate on newer Sandboxie versions. Thank you for your contribution!


Nothing has changed. The problem still exists. When I log into my Gmail accounts in any browser, they never log out on their own after closing the browser. (Cookies are enabled.) However, when I run any browser in Sandboxie, the Gmail accounts that are already logged in only stay logged in until I close the browser. After that, when I run the browser outside of Sandboxie, all Gmail accounts are logged out. Still only the Gmail accounts are logged out, forums, social media sites are not affected.

**Comment by love-code-yeyixiao** [comment]
Please try terminating sandboxed browser instead of closing in a normal way,and then see if the issue happen again.

**Comment by W1K23** [comment]
> As it has been 3 months since the last activity, we are automatically closing this issue in 14 days. If it is still present, please respond to help us investigate on newer Sandboxie versions. Thank you for your contribution!

Yes, the issue is still present on every computer in the family. It happens in Vivaldi and Firefox browsers when they are run in Sandboxie. The Gmail accounts never log out normally. They always stay signed in. However when I run the browsers in Sboxie, after 1-2 hours or whenever I close them, all Gmail accounts are logged out. This doesn't happen unless I clear the Sandboxie contents, which we do every few days. It's strange that only the Gmail accounts get logged out, even though we are signed into multiple other websites and those stay logged in. I still haven’t been able to fix the issue and I’ve tried everything.

**Comment by W1K23** [comment]
Yes, the issue is still present..........................

**Comment by offhub** [comment]
Have you ever tried giving direct access to browser cookies? If so, did you experience the same issue? If not, the issue is most likely caused by server-side session control.

The sandboxed browser copies the cookie file from the normal browser's profile to the sandbox and then uses it (i.e. they use the same cookie). As the content of the sandboxed cookie changes, the session information for this cookie on the server side may also change. Therefore, the session information of the normal and sandboxed cookies will no longer match, so logging in from the normal browser may no longer be successful.

##### Direct access to browser cookies

1. Backup your browser profile. (host)
2. Configure the sandbox to directly access browser cookies. (using templates)
3. Check if the same issue occurs with this configuration.

---

### TL;DR  

#### Using Same Cookies:  
- Logging out in the **sandboxed browser** prompts a login in the normal browser on the next use.  
- Logging out in the **normal browser** requires login in the sandboxed browser on the next use.  

#### Using Different Cookies:  
- Logging out in either the **sandboxed** or **normal** browser does not affect Gmail login status in the other environment, maintaining independent sessions.

> [!NOTE]
> Tests were done with both **gmail** and **github**.

---

### According to my tests:

#### Using same cookies

##### By logging out of the sandboxed browser

1. (Normal): I opened the browser, logged in to Gmail and then closed it.
2. (Sandboxed): I opened the browser in a clean sandbox and closed it.
   - Gmail login was successful.
3. (Normal): I opened the browser and closed it.
   - Gmail login was successful.
4. (Sandboxed): I opened the browser, logged out of Gmail and closed it.
5. (Normal): I opened the browser.
   - When the Gmail page opened, it asked me to log in again.

##### By logging out of the normal browser

1. (Normal): I opened the browser, logged in to Gmail and then closed it.
2. (Sandboxed): I opened the browser in a clean sandbox and closed it.
   - Gmail login was successful.
3. (Normal): I opened the browser and closed it.
   - Gmail login was successful.
4. (Sandboxed): I opened the browser and closed it.
   - Gmail login was successful.
5. (Normal): I opened the browser, logged out of Gmail and closed it.
6. (Sandboxed): I opened the browser.
   - When the Gmail page opened, it asked me to log in again.

#### Using different cookies

##### By logging out of the sandboxed browser

1. (Normal): I opened the browser, logged out from Gmail and then closed it.
2. (Sandboxed): I opened the browser, logged in to Gmail and then closed it.
3. (Normal): I opened the browser, logged in to Gmail and then closed it.
4. (Sandboxed): I opened the browser and closed it.
   - Gmail login was successful.
5. (Normal): I opened the browser and closed it.
   - Gmail login was successful.
6. (Sandboxed): I opened the browser, logged out of Gmail and closed it.
7. (Normal): I opened the browser.
   - Gmail login was successful.

##### By logging out of the normal browser

1. (Normal): I opened the browser, logged out from Gmail and then closed it.
2. (Sandboxed): I opened the browser, logged in to Gmail and then closed it.
3. (Normal): I opened the browser, logged in to Gmail and then closed it.
4. (Sandboxed): I opened the browser and closed it.
   - Gmail login was successful.
5. (Normal): I opened the browser and closed it.
   - Gmail login was successful.
6. (Sandboxed): I opened the browser and closed it.
   - Gmail login was successful.
7. (Normal): I opened the browser, logged out from Gmail and then closed it.
8. (Sandboxed): I opened the browser.
   - Gmail login was successful.

**Comment by W1K23** [comment]
@offhub 
In my normal browser, cookies are enabled and I never delete them. (Vivaldi, Firefox, Chrome...) I log into the sites I use, such as FB, Insta, various forums and Gmail accounts. I don't open other websites in the normal browser, so for me, the browser remains "clean" without unnecessary or "trash" cookies. I only use the normal browser for updates.

In Sboxie I only allow access to bookmarks so I can save my favorites. I usually close the browser running in Sboxie every 4-5 days, which deletes all "junk" data. (Except for cookies from the sites I had previously logged into in the normal browser). After this, if I open the browser again in normal mode or Sboxie, all previously logged-in sites remain logged in! -Except for Gmail accounts.- However, if I close the browser running in Sboxie after about 1 day instead of 3-4 days, the Gmail accounts also stay logged in. I can close and reopen it 100 times within 1 day and the Gmail accounts will still remain logged in.

The test you described works the same way for me if I perform it within 1 day. Gmail accounts are not logged out immediately by Sboxie but instead by the next day. I’ve tested this on multiple computers, different operating systems and various browsers, and it has been behaving this way for about a year and a half. Before that Gmail accounts would remain logged in for months without issue. 

It seems like this might be a Gmail issue. It’s possible that for security reasons, Gmail refreshes its cookies daily(?) and since there’s no sync between the refreshed cookie and the old (1 day) cookie, it logs it out. I can’t think of any other explanation. In a normal browser it never logs out of the accounts unless I put it into Sboxie.
I just don’t know what the solution would be to allow only Gmail cookies to refresh in Sboxie. Other websites don’t seem to require this.

**Comment by W1K23** [comment]
It is still present!

**Comment by Incog5** [comment]
Not sure if this is related or will help, but I had issues in Chrome related to getting signed out of the Chrome Browser Sync + gmail and some other sites, plus syncing being paused. But it didn't effect some other sites.  Cookies where unable to get consistently read, unless the logins were done in the sandboxed browser and never closing the windows and never updating chrome.  Debugged it by using a simple cookie demo site https://progressier.com/pwa-capabilities/cookies-demo.  Saved the cookie unsandboxed on this site. And then it wasn't appearing sandboxed when clicking "get" . But html5 local storage was being read just fine without issue sandboxed.

While fixing another issue which I found here, this fix also fixed the login/cookie issue:
https://github.com/sandboxie-plus/Sandboxie/issues/5180
Disabling "Chrome's elevation service".

I think Chrome detected some tampering when it was being launched sandboxed with the current setup I have, like some program is trying to read the cookies so it reset some sessions.

Another possible thread of interested might be:
https://forum.xanasoft.com/threads/paused-chrome-profile-sign-in-again.12463/
Device Bound Session Credentials (DBSC) started in chrome ver 142, so the sessions can be linked to hardware. But you were having issues on Firefox which hasn't implemented this.
